specutils/analysis/correlation.py

```diff
@@ -0,0 +1,119 @@
+import numpy as np
+
+import astropy.units as u
+from astropy import constants as const
+
+
+from ..manipulation import FluxConservingResampler
+
+
+def _normalize(observed_spectrum, template_spectrum):
+    """
+    Calculate a scale factor to be applied to the template spectrum so the
+    total flux in both spectra will be the same.
+
+    Parameters
+    ----------
+    observed_spectrum : :class:`~specutils.Spectrum1D`
+        The observed spectrum.
+    template_spectrum : :class:`~specutils.Spectrum1D`
+        The template spectrum, which needs to be normalized in order to be
+        compared with the observed spectrum.
+
+    Returns
+    -------
+    `float`
+        A float which will normalize the template spectrum's flux so that it
+        can be compared to the observed spectrum.
+    """
+    num = np.nansum((observed_spectrum.flux*template_spectrum.flux)/
+                 (observed_spectrum.uncertainty.array**2))
+    denom = np.nansum((template_spectrum.flux/
+                    observed_spectrum.uncertainty.array)**2)
+
+    return num/denom
+
+
+def template_correlate(observed_spectrum, template_spectrum):
```

Comment:
I just watched this video last night [Stop Using Classes](https://www.youtube.com/watch?v=o9pEzgHorH0). This suggestion sets off alarm bells.

Reply:
That's worth considering, @hcferguson, but we had that discussion in #266 .  To be fair I think we said we'd allow simple functions that we might wrap in classes later, so that doesn't have to block this PR, I suppose though.